supplementary_style_guide/style_guidelines/specific-doc-types.adoc

```diff
@@ -5,5 +5,104 @@
+
+* Enhancement
+* Bug Fix
+* Known Issue
+* Technology Preview
+* Deprecated Functionality
+
+Use the following general structure for consistency.
+
+Enhancement::
+ 
+Use present tense in one of the following formats:
+----
+This enhancement <present tense explanation>.
+----
+----
+With this update, <present tense explanation>.
+----
+.Example doc texts:
+----
+This enhancement optimizes migration of an RBD volume from one Cinder back end to another when the volume resides within the same Ceph cluster. If both volumes are in the same Ceph cluster, Ceph performs data migration instead of the cinder-volume process. This reduces migration time.
+----
+----
+With this update, you can create application credentials to use keystone to authenticate applications.
+----
+
+Bug Fix::
+
+Use past tense for the problem and present tense for the solution, in the following format:
+----
+Before this update,  <X problem> caused <Y situation> (OPTIONAL: under the following <Z conditions>). With this update, <fix> resolves the issue (OPTIONAL: and <agent> can <perform operation> successfully).
+----
+.Example doc text:
+----
+Before this update, the loopback device for Cinder iSCSI/LVM backend was not recreated after a system restart, which prevented the cinder-volume service from restarting. With this update, a systemd service recreates the loopback device and the Cinder iSCSI/LVM backend persists after a restart.
+----
+
+Known Issue::
+
+Use present tense for the issue and the imperative form for the workaround in the following format:
+----
+There is currently a known issue <present tense explanation> under <X conditions>. 
+
+Workaround: <workaround>.
+----
+.Example doc text:
+----
+Currently, you cannot use Orchestration (heat) templates with the director to deploy an overcloud that requires NFS as an Image service (glance) back end. There is currently no workaround for this issue.
+----
+----
+The Compute services (nova) might fail to deploy because the `nova_wait_for_compute_service` script is unable to query the Nova API. If a remote container image registry is used outside of the undercloud, the Nova API service might not finish deploying in time. 
+Workaround: Rerun the deployment command, or use a local container image registry on the undercloud.
+----
+
+Technology Preview::
+
+Use present tense in the following format:
+----
+In <release>, a Technology Preview is available for <feature explanation>.
+----
+.Example doc text: 
+----
+In Red Hat OpenStack Platform 16.0, a Technology Preview is available for the Bare Metal Provisioning service (ironic) deployed on an IPv6 provisioning network for BMaaS (Bare Metal as-a-Service) tenants.
+----
+
+Deprecated Functionality::
+Warn users about the following deprecation stages:
+
+* Plan to deprecate
+* Deprecate
+* Plan to remove
+* Remove 
+
+If available, inform users of alternative capabilities and workarounds. Use the following format:
+
+Deprecation notice::
```

Comment:
This uses the same term syntax (Deprecation notice::) as the main "Deprecated Functionality" heading. This should be some other type of heading so that it can be known that it falls under "Deprecated Functionality". 

Also, the examples for each should be directly under the format example blocks.

Maybe it would also help to have the titles for the example code blocks say which type it's for. Like "Example deprecation notice doc text" and "Example removal notice doc text" and "Example enhancement doc text", etc.